pinning/hardware/netio_addon.m4

```diff
@@ -53,8 +53,11 @@ ifdef(`conf_DCF77', `dnl
   pin(DCF1_PON, PA1, OUTPUT)
 ')dnl
 
-ifelse(value_HD44780_CONNECTION,`HD44780_I2CSUPPORT',`dnl
+ifelse(value_HD44780_TYPE,`HD44780_WDC2704',`dnl
```

Comment:
HD44780_WDC2704 and I2CSUPPORT should be one of many valid combinations only - defining the mapping that way results in wrong mapping if the display is connected directly.